client/src/modules/reports/generate/cash_report/cash_report.config.js

```diff
@@ -0,0 +1,74 @@
+angular.module('bhima.controllers')
+  .controller('cash_reportController', CashReportConfigController);
+
+CashReportConfigController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'reportData', '$state', 'CashboxService',
+];
+
+function CashReportConfigController($sce, Notify, SavedReports, AppCache, reportData, $state, Cashbox) {
+  var vm = this;
+  var cache = new AppCache('configure_cash_report');
+  var reportUrl = 'reports/finance/cash_report';
+
+  vm.previewGenerated = false;
+  vm.reportDetails = {};
+  vm.reportTypes = [
+    { id: 1, label: 'FORM.LABELS.ENTRY_EXIT' },
+    { id: 2, label: 'FORM.LABELS.ENTRY' },
+    { id: 3, label: 'FORM.LABELS.EXIT' },
+  ];
+
+  checkCachedConfiguration();
+
+  Cashbox.read(null, { detailed: 1 })
+    .then(function (cashboxes) {
+      cashboxes.forEach(function (cashbox) {
+        cashbox.hrlabel = cashbox.label + ' ' + cashbox.symbol;
+      });
+
+      vm.cashboxes = cashboxes;
+    })
+    .catch(Notify.handleError);
+
+  vm.clearPreview = function clearPreview() {
+    vm.previewGenerated = false;
+    vm.previewResult = null;
+  };
+
+  vm.preview = function preview(form) {
+    if (form.$invalid) { return; }
+
+    vm.reportDetails.account_id = vm.reportDetails.cashbox.account_id;
+    delete vm.reportDetails.cashbox;
```

Comment:
This `delete` statement makes the validation get triggered for the form.  If you fill out a report, then cancel the previous, it will be red and angry.